plugins/catalog-backend/src/modules/core/PlaceholderProcessor.test.ts

```diff
@@ -377,6 +374,82 @@ describe('PlaceholderProcessor', () => {
 
     expect(reader.readUrl).not.toHaveBeenCalled();
   });
+
+  it('not resolves relative file path with absolute path', async () => {
+    reader.readUrl.mockResolvedValue({
+      buffer: jest.fn().mockResolvedValue(Buffer.from('TEXT', 'utf-8')),
+    });
+    const processor = new PlaceholderProcessor({
+      resolvers: { text: textPlaceholderResolver },
+      reader,
+      integrations,
+    });
+
+    await expect(
+      processor.preProcessEntity(
+        {
+          apiVersion: 'a',
+          kind: 'k',
+          metadata: { name: 'n' },
+          spec: {
+            data: {
+              $text: '/tmp/catalog-info.yaml',
+            },
+          },
+        },
+        {
+          type: 'url',
+          target: './a/b/catalog-info.yaml',
+        },
+        () => {},
+      ),
+    ).rejects.toThrow(
+      /^Placeholder \$text could not form a URL out of \.\/a\/b\/catalog-info\.yaml and \/tmp\/catalog-info\.yaml, Error: URL parsing failed\./,
+    );
+
+    expect(reader.readUrl).not.toHaveBeenCalled();
+  });
+
+  it('resolved relative file path with same directory path', async () => {
+    reader.readUrl.mockResolvedValue({
+      buffer: jest.fn().mockResolvedValue(Buffer.from('TEXT', 'utf-8')),
+    });
+    const processor = new PlaceholderProcessor({
+      resolvers: { text: textPlaceholderResolver },
+      reader,
+      integrations,
+    });
+
+    await expect(
+      processor.preProcessEntity(
+        {
+          apiVersion: 'a',
+          kind: 'k',
+          metadata: { name: 'n' },
+          spec: {
+            data: {
+              $text: './c/catalog-info.yaml',
```

Comment:
should definitely add some actively malicious path tests that try to exit from the path that the entity data is in, verifying that it seems impossible to "escape" the directory